### PR TITLE
Add support for users to register

### PR DESCRIPTION
**Pull request by ifeomi**
- When a user signs in, if they haven't previously registered, a modal pops up for them to provide a username which is stored 
- There is client-side and server-side validation to make sure a username is submitted and that it hasn't already been taken

image

### CODE REVIEW

**Review by lalaAdasia00 — APPROVED** [review]
This is so cool!

**Review by nicholashusin — APPROVED** [review]
LGTM :)

File: routes/register.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+'use strict';
+const datastore = require('../modules/datastore.js');
+const multer  = require('multer');
+
+const USER_KIND = "User";
+
+async function isValid(username) {
```
Comment (chronologos) [inline routes/register.js:8]: other interesting validity checks:
- empty string
- string with just spaces
etc.

sanitization is another related area that we could look into [[1](https://linguinecode.com/post/validate-sanitize-user-input-javascript)]
e.g. if the user enters something like `foo').delete()` as their username will it be parsed into a string correctly

**Review by chronologos — APPROVED** [review]

File: routes/register.js (hunk 1 above)
Comment (ifeomi) [reply routes/register.js:8]: thank you for these suggestions! i created a ticket to add more robust validation and i'll add a todo.

**Pull request merged**